pkg/server/config/config_test.go

```diff
@@ -194,3 +195,103 @@ func TestConfigCompile(t *testing.T) {
 		})
 	}
 }
+
+func TestRuleEval(t *testing.T) {
+	t.Parallel()
+
+	token := map[string]any{
+		"jti":                   "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
+		"sub":                   "repo:abcxyz/test:ref:refs/heads/main",
+		"aud":                   "https://github.com/abcxyz",
+		"ref":                   "refs/heads/main",
+		"sha":                   "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+		"repository":            "abcxyz/test",
+		"repository_owner":      "abcxyz",
+		"repository_owner_id":   "111111111",
+		"run_id":                "1111111111",
+		"run_number":            "11",
+		"run_attempt":           "1",
+		"repository_visibility": "private",
+		"repository_id":         "111111111",
+		"actor_id":              "1111111",
+		"actor":                 "test",
+		"workflow":              "Test",
+		"head_ref":              "",
+		"base_ref":              "",
+		"event_name":            "workflow_dispatch",
+		"ref_type":              "branch",
+		"workflow_ref":          "abcxyz/test/.github/workflows/test.yaml@refs/heads/main",
+		"workflow_sha":          "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+		"job_workflow_ref":      "abcxyz/test/.github/workflows/test.yaml@refs/heads/main",
+		"job_workflow_sha":      "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+		"iss":                   "https://token.actions.githubusercontent.com",
+		"nbf":                   "1669925693",
+		"exp":                   "1669926893",
+	}
+
+	cases := []struct {
+		name      string
+		rule      *Rule
+		token     map[string]interface{}
+		want      bool
+		expErr    bool
+		expErrMsg string
+	}{
+		{
+			name: "success",
+			rule: &Rule{
+				If: "assertion.workflow == 'Test'",
+			},
+			token:     token,
+			want:      true,
+			expErr:    false,
+			expErrMsg: "",
+		},
+		{
+			name: "no match",
+			rule: &Rule{
+				If: "assertion.workflow == 'not valid'",
+			},
+			token:     token,
+			want:      false,
+			expErr:    false,
+			expErrMsg: "",
+		},
+		{
+			name: "bad assertion",
+			rule: &Rule{
+				If: "assertion.doesntexist == 'not valid'",
+			},
+			token:     token,
+			want:      false,
+			expErr:    true,
+			expErrMsg: "failed to evaluate CEL expression: no such key: doesntexist",
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
```

Comment:
I believe starting at Go 1.22. You don't have to do this anymore. https://tip.golang.org/doc/go1.22

Reply:
ack